orderbook-rs/src/gasoracle/mod.rs

```diff
@@ -0,0 +1,30 @@
+use ethers::middleware::gas_oracle::GasCategory;
+use ethers::prelude::gas_oracle::blocknative::Response as BlockNativeResponse;
+use reqwest::{header::AUTHORIZATION, Client};
+
+/// Blocknative Gas Oracle.
+/// Returns max priority fee and max fee from blocknative api.
+///
+/// # Arguments
+/// * `api_key` - Optional blocknative api key.
+/// * `chain_id` - Network Chain Id.
+///
+pub async fn gas_price_oracle(
+    api_key: Option<String>,
+    chain_id: u64,
+) -> anyhow::Result<(f64, f64)> {
+    let client = Client::new();
+    let url = format!(
+        "{}{}",
+        "https://api.blocknative.com/gasprices/blockprices?chainid=", chain_id
```

Comment:
magic strings like this should probably be a constant, especially if we plan to ever hit more than one endpoint for an API

also there is a url crate fyi https://docs.rs/url/latest/url/

Reply:
Resolved.